### DISCUSSION TITLE
Noobaa core / pg HA

### DISCUSSION THREAD

**Discussion by Alansyf** [noobaa/noobaa-operator#1165] [Q&A]
Hi, 

We would like to use noobaa as our productive object gateway. 

1. Statefulset noobaa-core and noobaa-db-pg can be scaled to 3 pods for HA purpose? 
2. Is this number can be passed when install noobaa?

**Comment by nimrod-becker** [comment]
Hi [Alansyf](https://github.com/Alansyf),

While its true that the DB is the actual state of the system, we currently don't support more than 1 core. For the DB, simply scaling won't do it and you'd need a solution on the DB level to support more than 1 replica.

We are in the process of designing our HA solution for the DB based on the psql replica, but it would take some time.
I am wondering if providing a way to connect to a psql DB which is external to NooBaa, so essentially managed by you would be a good solution for you? It means you can set up a 3 replica psql and NooBaa will connect to it. We don't have that as well, but it might be feasible to provide this faster then the fully managed db HA (which we'll do anyway)

**Reply by Alansyf** [reply]
Hi @nimrod-becker , 
It will be ok we build up a fresh HA postgresql on kubernates, and we just provide the connections string etc to a conf file when install noobaa operator. This looks good for most of people I believe. 

For scaling out the core, I am also ok if this is something in your roadmap. 
As long as we have clear way to upgrade noobaa as I asked in ticket [1164](https://github.com/noobaa/noobaa-operator/discussions/1164)

**Comment by nimrod-becker** [comment]
I believe we started with the option to support external DN
@dannyzaken to chime in

**Reply by dannyzaken** [reply]
@Alansyf We currently don't have a way to connect to an external DB for postgres
